sdk/core/azure-core/src/main/java/com/azure/core/http/rest/SyncRestProxy.java

```diff
@@ -0,0 +1,692 @@
+                        sink.error(new UnexpectedLengthException(String.format(BODY_TOO_SMALL,
+                            currentTotalLength[0], expectedLength), currentTotalLength[0], expectedLength));
+                    } else {
+                        sink.complete();
+                    }
+                    return;
+                }
+
+                currentTotalLength[0] += buffer.remaining();
+                if (currentTotalLength[0] > expectedLength) {
+                    sink.error(new UnexpectedLengthException(String.format(BODY_TOO_LARGE,
+                        currentTotalLength[0], expectedLength), currentTotalLength[0], expectedLength));
+                    return;
+                }
+
+                sink.next(buffer);
+            });
+        });
+    }
+
+    static BinaryData validateLengthSync(final HttpRequest request) {
+        final BinaryData binaryData = request.getContent();
+        if (binaryData == null) {
+            return binaryData;
+        }
+
+        final long expectedLength = Long.parseLong(request.getHeaders().getValue("Content-Length"));
+
+        long length = binaryData.getLength();
+
+        if (length > expectedLength) {
+            throw new UnexpectedLengthException(String.format(BODY_TOO_LARGE,
+                binaryData.getLength(), expectedLength), binaryData.getLength(), expectedLength);
+        }
+        return binaryData;
+    }
+
+    /**
+     * Starts the tracing span for the current service call, additionally set metadata attributes on the span by passing
+     * additional context information.
+     *
+     * @param method Service method being called.
+     * @param context Context information about the current service call.
+     * @return The updated context containing the span context.
+     */
+    private Context startTracingSpan(Method method, Context context) {
+        // First check if tracing is enabled. This is an optimized operation, so it is done first.
+        if (!TracerProxy.isTracingEnabled()) {
+            return context;
+        }
+
+        // Then check if this method disabled tracing. This requires walking a linked list, so do it last.
+        if ((boolean) context.getData(Tracer.DISABLE_TRACING_KEY).orElse(false)) {
+            return context;
+        }
+
+        String spanName = interfaceParser.getServiceName() + "." + method.getName();
+        context = TracerProxy.setSpanName(spanName, context);
+        return TracerProxy.start(spanName, context);
+    }
+
+    /**
+     * Create a HttpRequest for the provided Swagger method using the provided arguments.
+     *
+     * @param methodParser the Swagger method parser to use
+     * @param args the arguments to use to populate the method's annotation values
+     * @return a HttpRequest
+     * @throws IOException thrown if the body contents cannot be serialized
+     */
+    private HttpRequest createHttpRequest(SwaggerMethodParser methodParser, Object[] args) throws IOException {
```

Comment:
Outside of this PR. At some point we should figure out how to share code between sync and async proxy. But I think duplicating the logic in the branch is fine for now - allows easy sync from main.

Reply:
I may want to go against waiting for after this PR to determine all the async methods that can become sync and shared. If we wait until later we may build on what is here making it harder to untangle